### PR TITLE
RATIS-1028. [Checkstyle] Enable AvoidStarImport

### PR DESCRIPTION
**Pull request by amaliujia**
## What changes were proposed in this pull request?

enable AvoidImportStar in checksytle config  and fix existing `import *`.

## What is the link to the Apache JIRA
https://issues.apache.org/jira/browse/RATIS-1028

## How was this patch tested?
UT

### CODE REVIEW

**Comment by runzhiwang** [comment]
+1

**Comment by amaliujia** [comment]
@szetszwo 

I am wondering if you can commit this. If we have agreed to enable `avoid import start`, the earlier we enable it, the better.

File: ratis-common/src/main/java/org/apache/ratis/protocol/RaftGroup.java [hunk 1]
```diff
@@ -35,7 +40,8 @@ public static RaftGroup emptyGroup() {
 
   /** @return a group with the given id and peers. */
   public static RaftGroup valueOf(RaftGroupId groupId, RaftPeer... peers) {
-    return new RaftGroup(groupId, peers == null || peers.length == 0? Collections.emptyList(): Arrays.asList(peers));
+    return new RaftGroup(groupId, peers == null || peers.length == 0? Collections.emptyList(): Arrays
+        .asList(peers));
```
Comment (szetszwo) [inline ratis-common/src/main/java/org/apache/ratis/protocol/RaftGroup.java:44]: Please revert this whitespace change.  We allow 120 characters.

File: ratis-grpc/src/main/java/org/apache/ratis/grpc/client/GrpcClientStreamer.java [hunk 2]
```diff
@@ -17,23 +17,38 @@
  */
 package org.apache.ratis.grpc.client;
 
+import java.util.Collection;
+import java.util.Deque;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Objects;
```
Comment (szetszwo) [inline ratis-grpc/src/main/java/org/apache/ratis/grpc/client/GrpcClientStreamer.java:24]: Why move java.util away from other java imports?

File: ratis-common/src/main/java/org/apache/ratis/util/FileUtils.java [hunk 3]
```diff
@@ -17,6 +17,12 @@
  */
 package org.apache.ratis.util;
 
+import java.nio.file.FileVisitResult;
+import java.nio.file.Files;
+import java.nio.file.LinkOption;
+import java.nio.file.Path;
+import java.nio.file.SimpleFileVisitor;
+import java.nio.file.StandardOpenOption;
```
Comment (szetszwo) [inline ratis-common/src/main/java/org/apache/ratis/util/FileUtils.java:25]: Why move java.util away from other java imports?

File: ratis-examples/src/main/java/org/apache/ratis/examples/filestore/FileStoreCommon.java [hunk 4]
```diff
@@ -18,11 +18,14 @@
...
 import org.apache.ratis.thirdparty.com.google.protobuf.ByteString;
-import org.apache.ratis.util.*;
 
 import java.io.IOException;
 import java.nio.file.Path;
 import java.util.concurrent.CompletableFuture;
+import org.apache.ratis.util.JavaUtils;
+import org.apache.ratis.util.ProtoUtils;
+import org.apache.ratis.util.SizeInBytes;
+import org.apache.ratis.util.TraditionalBinaryPrefix;
```
Comment (szetszwo) [inline ratis-examples/src/main/java/org/apache/ratis/examples/filestore/FileStoreCommon.java:28]: Why move org.apache.ratis.util away from other org.apache.ratis imports?

File: ratis-grpc/src/main/java/org/apache/ratis/grpc/server/GrpcLogAppender.java [hunk 5]
```diff
@@ -17,6 +17,12 @@
  */
 package org.apache.ratis.grpc.server;
 
+import java.util.LinkedList;
+import java.util.Map;
+import java.util.Objects;
+import java.util.Optional;
+import java.util.Queue;
+import java.util.UUID;
```
Comment (szetszwo) [inline ratis-grpc/src/main/java/org/apache/ratis/grpc/server/GrpcLogAppender.java:25]: Why move java.util away from other java imports?

File: ratis-logservice/src/main/java/org/apache/ratis/logservice/api/LogServiceClient.java [hunk 6]
```diff
@@ -25,16 +25,23 @@
...
-import org.apache.ratis.protocol.*;
 
 import java.io.IOException;
 import java.util.List;
 import java.util.Set;
 import java.util.stream.Collectors;
+import org.apache.ratis.protocol.ClientId;
+import org.apache.ratis.protocol.RaftClientReply;
+import org.apache.ratis.protocol.RaftGroup;
+import org.apache.ratis.protocol.RaftPeer;
```
Comment (szetszwo) [inline ratis-logservice/src/main/java/org/apache/ratis/logservice/api/LogServiceClient.java:44]: This should move to next to the other org.apache.ratis imports.

File: ratis-netty/src/main/java/org/apache/ratis/netty/client/NettyClientRpc.java [hunk 7]
```diff
@@ -30,6 +29,14 @@
...
 
 import java.io.IOException;
+import org.apache.ratis.protocol.ClientId;
+import org.apache.ratis.protocol.GroupInfoRequest;
+import org.apache.ratis.protocol.GroupListRequest;
+import org.apache.ratis.protocol.GroupManagementRequest;
+import org.apache.ratis.protocol.RaftClientReply;
+import org.apache.ratis.protocol.RaftClientRequest;
+import org.apache.ratis.protocol.RaftPeerId;
+import org.apache.ratis.protocol.SetConfigurationRequest;
```
Comment (szetszwo) [inline ratis-netty/src/main/java/org/apache/ratis/netty/client/NettyClientRpc.java:39]: This should move to next to the other org.apache.ratis imports.

File: ratis-server/src/main/java/org/apache/ratis/server/impl/LeaderState.java [hunk 8]
```diff
@@ -17,9 +17,27 @@
...
 
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+import java.util.Optional;
```
Comment (szetszwo) [inline ratis-server/src/main/java/org/apache/ratis/server/impl/LeaderState.java:28]: This should move to next to the other java imports.

File: ratis-server/src/main/java/org/apache/ratis/server/impl/LogAppender.java [hunk 9]
```diff
@@ -17,7 +17,20 @@
  */
 package org.apache.ratis.server.impl;
 
+import java.util.Collections;
+import java.util.Iterator;
+import java.util.List;
+import java.util.NoSuchElementException;
+import java.util.Objects;
+import java.util.UUID;
```
Comment (szetszwo) [inline ratis-server/src/main/java/org/apache/ratis/server/impl/LogAppender.java:25]: This should move to next to the other java imports.

File: ratis-server/src/main/java/org/apache/ratis/server/impl/RaftConfiguration.java [hunk 10]
```diff
@@ -17,11 +17,15 @@
  */
 package org.apache.ratis.server.impl;
 
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.List;
+import java.util.Objects;
```
Comment (szetszwo) [inline ratis-server/src/main/java/org/apache/ratis/server/impl/RaftConfiguration.java:24]: This should move to next to the other java imports.

File: ratis-server/src/main/java/org/apache/ratis/server/impl/RaftServerImpl.java [hunk 11]
```diff
@@ -17,9 +17,57 @@
...
 
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.List;
+import java.util.Objects;
+import java.util.Optional;
+import java.util.OptionalLong;
+import java.util.Set;
```
Comment (szetszwo) [inline ratis-server/src/main/java/org/apache/ratis/server/impl/RaftServerImpl.java:28]: This should move to next to the other java imports.

File: ratis-server/src/main/java/org/apache/ratis/server/impl/RaftServerImpl.java [hunk 12]
```diff
@@ -515,7 +571,7 @@ NotLeaderException generateNotLeaderException() {
     return new NotLeaderException(getMemberId(), conf.getPeer(leaderId), peers);
   }
 
-  private LifeCycle.State assertLifeCycleState(Set<LifeCycle.State> expected) throws ServerNotReadyException {
+  private LifeCycle.State assertLifeCycleState(Set<State> expected) throws ServerNotReadyException {
```
Comment (szetszwo) [inline ratis-server/src/main/java/org/apache/ratis/server/impl/RaftServerImpl.java:574]: Let's keep using LifeCycle.State.

File: ratis-server/src/main/java/org/apache/ratis/server/impl/RaftServerProxy.java [hunk 13]
```diff
@@ -17,6 +17,13 @@
  */
 package org.apache.ratis.server.impl;
 
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.CompletionException;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.ConcurrentMap;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+import java.util.concurrent.TimeUnit;
```
Comment (szetszwo) [inline ratis-server/src/main/java/org/apache/ratis/server/impl/RaftServerProxy.java:26]: This should move to next to the other java imports.

File: ratis-server/src/main/java/org/apache/ratis/server/impl/ServerState.java [hunk 14]
```diff
@@ -18,15 +18,20 @@
...
 import org.apache.ratis.server.RaftServerConfigKeys;
 import org.apache.ratis.server.protocol.TermIndex;
 import org.apache.ratis.server.raftlog.RaftLog;
 import org.apache.ratis.server.raftlog.memory.MemoryRaftLog;
 import org.apache.ratis.server.raftlog.segmented.SegmentedRaftLog;
-import org.apache.ratis.server.storage.*;
 import org.apache.ratis.proto.RaftProtos.InstallSnapshotRequestProto;
 import org.apache.ratis.proto.RaftProtos.LogEntryProto;
+import org.apache.ratis.server.storage.RaftStorage;
+import org.apache.ratis.server.storage.SnapshotManager;
```
Comment (szetszwo) [inline ratis-server/src/main/java/org/apache/ratis/server/impl/ServerState.java:34]: Let's put the  org.apache.ratis.server together.

File: ratis-server/src/main/java/org/apache/ratis/server/impl/StateMachineUpdater.java [hunk 15]
```diff
@@ -52,7 +56,7 @@
  * asynchronously so that this will not block normal raft protocol.
  *
  * If the auto log compaction is enabled, the state machine updater thread will
- * trigger a snapshot of the state machine by calling
+ * trigger a snapshot of the state machine by callingestRestartFollower/
```
Comment (szetszwo) [inline ratis-server/src/main/java/org/apache/ratis/server/impl/StateMachineUpdater.java:59]: Why changing this?

File: ratis-server/src/main/java/org/apache/ratis/server/raftlog/segmented/SegmentedRaftLogCache.java [hunk 16]
```diff
@@ -17,6 +17,13 @@
  */
 package org.apache.ratis.server.raftlog.segmented;
 
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.Iterator;
+import java.util.List;
+import java.util.NoSuchElementException;
+import java.util.Objects;
```
Comment (szetszwo) [inline ratis-server/src/main/java/org/apache/ratis/server/raftlog/segmented/SegmentedRaftLogCache.java:26]: This should move to next to the other java imports.

File: ratis-server/src/main/java/org/apache/ratis/server/raftlog/segmented/SegmentedRaftLogReader.java [hunk 17]
```diff
@@ -17,6 +17,16 @@
...
+import java.io.BufferedInputStream;
+import java.io.Closeable;
+import java.io.DataInputStream;
+import java.io.EOFException;
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.FileNotFoundException;
+import java.io.FilterInputStream;
+import java.io.IOException;
+import java.io.InputStream;
```
Comment (szetszwo) [inline ratis-server/src/main/java/org/apache/ratis/server/raftlog/segmented/SegmentedRaftLogReader.java:29]: This should move to next to the other java imports.

File: ratis-server/src/main/java/org/apache/ratis/server/storage/MetaFile.java [hunk 18]
```diff
@@ -17,12 +17,18 @@
  */
 package org.apache.ratis.server.storage;
 
+import java.io.BufferedReader;
+import java.io.BufferedWriter;
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.IOException;
+import java.io.InputStreamReader;
+import java.io.OutputStreamWriter;
```
Comment (szetszwo) [inline ratis-server/src/main/java/org/apache/ratis/server/storage/MetaFile.java:26]: This should move to next to the other java imports.

File: ratis-server/src/main/java/org/apache/ratis/server/storage/RaftStorageDirectory.java [hunk 19]
```diff
@@ -17,6 +17,12 @@
  */
 package org.apache.ratis.server.storage;
 
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Comparator;
+import java.util.List;
+import java.util.Objects;
+import java.util.UUID;
```
Comment (szetszwo) [inline ratis-server/src/main/java/org/apache/ratis/server/storage/RaftStorageDirectory.java:25]: This should move to next to the other java imports.

**Comment by amaliujia** [comment]
Thank you! 

Those changes are automatically done by my IntelliJ which has configured by CheckStyle, and I assumed checks pass means these changes are good to Ratis

I will take another look on your comments about java import ordering.  But I also agreed we should put more time on other areas. So if such checkstyle work needs too much time, I will postpone this change.

**Pull request closed without merging**